.github/workflows/validate.yml

```diff
@@ -66,27 +66,24 @@ jobs:
       GHC_FOR_RELEASE: ${{ format('["{0}"]', env.GHC_FOR_RELEASE) }}
     strategy:
       matrix:
-        os: ["ubuntu-latest", "macos-13", "windows-latest"]
+        os: ['ubuntu-latest', 'macos-13', 'windows-latest']
```

Comment:
I was in a rush with the macos runners fix and didn't study the matter. It looks like Andreas revamped quoting in a peculiar way: most of the time it's single quotes but in this particular case, with the `os` key, it's no quotes at all. Of course, I trust Andreas unconditionally. @geekosaur, you have been shepherding this PR, and maybe you're the right person to ask: do you think you could make it uniform with `master` and remove the quotes, if this makes sense?

Reply:
@andreasabel any particular reason why you removed the quotes around the values of `os` in #9886? Superficially it feels like quotes are always better than no quotes.

Reply:
Well, the thing is that actions are a mix of three syntaxes: YAML, javascript expressions, and often shell.
Each syntax has its quoting rules and I think it is least confusing if one tries to be aware which syntax one is dealing at each point and use the quoting rules of this syntax.  E.g. for strings:
- YAML: everything is a string, so no quoting, except when things look like lists or records or floats
- javascript: strings are usually in single quotes
- shell (bash): no quotes for plain strings, double quotes for strings that contain spaces or could contain spaces after variable expansion, single quotes for preventing expansion

In this particular case, we are dealing with yaml ordinary strings, so no quotes.

I tried to be very consistent, previously, the file was a mix of various quoting styles...

Reply:
I'm making it consistent with `master`; if we want to audit quoting, we should do it there and backport. I am also partially backporting #9953, to the extent that it added a comment to `validate.yml`.

Reply:
Thanks a lot, Andreas! This is super clear. I will copy your explanation to my little personal cookbook for future reference.